### PR TITLE
Add entrypoint for exercise information

### PR DESCRIPTION
**Pull request by erikmd**
**Kind:** feature

(Summary: as discussed with @yurug and @leunam217, this new entrypoint appeared to be necessary for the `learn-ocaml.el`-based student workflow)

This PR is ready, it subsumes and replaces #297.

To test it:

```bash
make
make opaminstall
learn-ocaml "--repo=$PWD/demo-repository" build serve
export TOKEN="…"
xdg-open "http://localhost:8080/description/demo#token=$TOKEN"
```

### CODE REVIEW

File: src/app/learnocaml_common.ml [hunk 1]
```diff
@@ -969,3 +969,271 @@ let setup_prelude_pane ace prelude =
...
+            Manip.appendChild exp (pcdata "[-]")
+          end
+        else
+          begin
+            Manip.removeChildren (find_component cid);
+            displayed := false;
+            Manip.removeChildren exp;
+            Manip.appendChild exp (pcdata "[+]")
+          end;
+        true
```
Comment (yurug) [inline src/app/learnocaml_common.ml]: I have the feeling that this code could be refactored to be more readable.

File: src/app/learnocaml_common.ml [hunk 2]
```diff
@@ -969,3 +969,271 @@ let setup_prelude_pane ace prelude =
...
+                                      focus in
+      let requirements =
+        opt ex_meta.Meta.requirements @@ fun requirements ->
+                                         [%i "Skills required:"],
+                                         display_list ~sep:(pcdata "") @@
+                                           List.map (fun s ->
+                                               display_skill_link
+                                                 (try SSet.elements (SMap.find s req_map) with Not_found -> [])
+                                                 "learnocaml-exo-requirements-meta" (`Requirements s))
+                                             requirements in
```
Comment (yurug) [inline src/app/learnocaml_common.ml]: Same here: there are some factorization opportunities that are missing. The introduction of well-designed combinators should help. Please remember that a good functional programming style is made of composing small and general functions... I know that the source code of learn-ocaml has not been initially written with that kind of stylistic consideration in mind but I would like to improve on that.

**Comment by erikmd** [comment]
Hi @yurug, thanks for your feedback! I agree that a code-factorization is welcome here :)

I have to go away from my keyboard now (but I should be able to do it tonight or tomorrow Monday)

**Comment by erikmd** [comment]
Dear @yurug 

> This PR seems fine but if you could do a pass to improve code factorization, it would be nice!

Done. Tested locally (just need to wait for the CI :)

**Comment by erikmd** [comment]
Dear @yurug, this PR #326 was already ready, but for more clarity I've just "split" the last commit into several parts:

* the commit ffe6040 deals with the refactoring you had requested in your review;
* the commit dacffc5 performs a further small refactoring by modifying an interface (but which seems unused elsewhere), of course I can revert that commit if you disagree;
* and the last two commits 26df32a and f737279 are just details (w.r.t. spaces/colon) entangled with an automatic update of `translations/fr.po` (thanks to `make update-fr-translation`, which needs the `gettext` package - and an editor such as emacs+https://melpa.org/#/po-mode)

**Comment by yurug** [comment]
Thank you @erikmd !

**Pull request merged**